tests/unit/python/foglamp/services/south/test_ingest.py

```diff
@@ -0,0 +1,517 @@
+
+        # WHEN
+        await Ingest.start(core_mgt_host=None, core_mgt_port=None)
+
+        # THEN
+        assert 1 == create_cfg.call_count
+        assert 1 == get_cfg.call_count
+        assert Ingest._stop is False
+        assert Ingest._started is True
+        assert Ingest._readings_list_size == int(Ingest._readings_buffer_size / (
+            Ingest._max_concurrent_readings_inserts))
+        assert Ingest._last_insert_time is 0
+        assert Ingest._max_concurrent_readings_inserts == len(Ingest._insert_readings_tasks)
+        assert Ingest._max_concurrent_readings_inserts == len(Ingest._insert_readings_wait_tasks)
+        assert Ingest._max_concurrent_readings_inserts == len(Ingest._readings_list_batch_size_reached)
+        assert Ingest._max_concurrent_readings_inserts == len(Ingest._readings_list_not_empty)
+        assert Ingest._max_concurrent_readings_inserts == len(Ingest._readings_lists)
+        assert 0 == log_warning.call_count
+
+    @pytest.mark.asyncio
+    async def test_stop(self, mocker):
+        # GIVEN
+        mocker.patch.object(StorageClient, "__init__", return_value=None)
+        mocker.patch.object(ReadingsStorageClient, "__init__", return_value=None)
+        log_exception = mocker.patch.object(ingest._LOGGER, "exception")
+        create_cfg = mocker.patch.object(ConfigurationManager, "create_category", return_value=mock_coro())
+        get_cfg = mocker.patch.object(ConfigurationManager, "get_category_all_items", return_value=get_cat(Ingest.default_config))
+        mocker.patch.object(Ingest, "_write_statistics", return_value=mock_coro())
+        mocker.patch.object(Ingest, "_insert_readings", return_value=mock_coro())
+
+        # WHEN
+        await Ingest.start(core_mgt_host=None, core_mgt_port=None)
+        await asyncio.sleep(1)
+        await Ingest.stop()
+
+        # THEN
+        assert 1 == create_cfg.call_count
+        assert 1 == get_cfg.call_count
+        assert Ingest._stop is True
+        assert Ingest._started is False
+        assert Ingest._insert_readings_wait_tasks is None
+        assert Ingest._insert_readings_tasks is None
+        assert Ingest._readings_lists is None
+        assert Ingest._readings_list_batch_size_reached is None
+        assert Ingest._readings_list_not_empty is None
+        assert Ingest._readings_lists_not_full is None
+        assert 0 == log_exception.call_count
+
+    @pytest.mark.asyncio
+    async def test_increment_discarded_readings(self, mocker):
+        # GIVEN
+        # WHEN
+        Ingest.increment_discarded_readings()
+
+        # THEN
+        assert 1 == Ingest._discarded_readings_stats
+
+    @pytest.mark.asyncio
+    async def test__insert_readings(self, mocker):
+        # TODO: This method uses a while True loop. Investigate as to how to write unit test for an infinite loop.
+        pass
+
+    @pytest.mark.asyncio
+    async def _write_statistics(self, mocker):
```

Comment:
this should be skipped with reason! and name `test__write_statistics`; 
its call is not covered in exposed def?